equinox-shipment/Domain/FinalizationTransactionProcessor.fs

```diff
@@ -0,0 +1,59 @@
+module FinalizationTransactionProcessor
+
+    type Service(transactions: FinalizationTransaction.Service, containers: Container.Service, shipments: Shipment.Service) =
+
+        // Transaction ID can be a md5 hash of all shipments ID
+        let execute (transactionId: string) : FinalizationTransaction.Events.Event option -> Async<bool> =
+            let rec loop (update: FinalizationTransaction.Events.Event option) =
+                async {
+                    let loop event = loop (Some event)
+
+                    let! action =
+                        transactions.Apply(transactionId, update)
+
+                    match action with
+                    | FinalizationTransaction.Action.AssignShipments (containerId, shipmentIds) ->
+                        let! result =
+                            shipmentIds
+                            |> Array.map (fun sId ->
+                                async {
+                                    let! res = shipments.Execute(sId, Shipment.Command.Assign containerId)
+                                    return (sId, res)
+                                }
+                            )
+                            |> Async.Parallel
```

Comment:
this needs throttling on Cosmos
clone https://github.com/jet/dotnet-templates/blob/a207ab9cacfa05a47399204cd2a2273b195cb392/propulsion-consumer/Infrastructure.fs#L21-L34 into an infrastructure.fs and take a Dop as an input in the ctor (something like 8 if your store will typically have a single and/or a low number of physical partition)

Reply:
If you don't mind, I'll add a comment here but won't be doing cosmos throttling in this sample.

Reply:
a TODO will be fine (I'll likely add it separately as I find it important to convey the concept)

Reply:
I've added a todo comment. I'll resolve this comment. Thanks!